numpy/lib/function_base.py

```diff
@@ -4791,15 +4793,15 @@ def delete(arr, obj, axis=None):
             pass
         else:
             slobj[axis] = slice(None, start)
-            new[slobj] = arr[slobj]
+            new[tuple(slobj)] = arr[tuple(slobj)]
```

Comment:
Probably going out of PR again, but one could define `slobj` as a prefix tuple and just add the relevant slice here.